pypeit/core/wavecal/waveio.py

```diff
@@ -107,9 +107,9 @@ def load_reid_arxiv(arxiv_file):
 
     """
     # ToDO put in some code to allow user specified files rather than everything in the main directory
-    calibfile = data.get_reid_arxiv_filepath(arxiv_file)
+    calibfile, in_cache = data.get_reid_arxiv_filepath(arxiv_file)
     # This is a hack as it will fail if we change the data model yet again for wavelength solutions
-    if calibfile[-4:] == 'json':
+    if calibfile[-4:] == 'json' or in_cache == 'json':
```

Comment:
I meant to add to the initial PR comment that this is an ugly hack for functionality, and that I will look to provide a more elegant solution to this (and any other `get_*_filepath()` bugs) in a regular PR to `develop`.

I completely agree about using `pathlib`, and use its features extensively in other projects.  It will be worth combing the repo for path issues like this -- but I would consider it outside the context of this hotfix.